Include/pylifecycle.h

```diff
@@ -21,17 +21,17 @@ PyAPI_FUNC(int) Py_SetStandardStreamEncoding(const char *encoding,
                                              const char *errors);
 
 /* PEP 432 Multi-phase initialization API (Private while provisional!) */
-PyAPI_FUNC(void) _Py_InitializeCore(const _PyCoreConfig *);
+PyAPI_FUNC(const char*) _Py_InitializeCore(const _PyCoreConfig *);
```

Comment:
I'm wondering if a `typedef char _Py_InitErrorMessage;` might make these API signatures a bit clearer:

    PyAPI_FUNC(const _Py_InitErrorMessage*) _Py_InitializeCore(const _PyCoreConfig *);

While I currently guessed "Oh, that will be an error message" in the context of the patch, I'm not sure I would have without that extra context.

Using a typedef also leaves the door open to upgrading to a struct later if we find a reason for doing that.

Reply:
I didn't see the need for a structure, but after converting more code to avoid calling Py_FatalError(), I found two places where we exit Python but don't want to exit with abort():

```
        if (Py_ReadHashSeed(seed_text, &use_hash_seed, &hash_seed) < 0) {
            Py_FatalError("PYTHONHASHSEED must be \"random\" or an integer "
                          "in range [0; 4294967295]");
        }
```

and

```
    m = PyImport_ImportModule("site");
    if (m == NULL) {
        fprintf(stderr, "Failed to import the site module\n");
        PyErr_Print();
        Py_Finalize();
        exit(1);
    }
```

These errors are more on the user side, they are not "bugs" of Python itself.

Reply:
So I implemented a new _PyInitError "type" (technically a structure) to indicate if it's an "user error" or another error. Moreover, I added the name of the function where the error was raised, to the error message.